crates/ide-db/src/defs.rs

```diff
@@ -24,6 +24,12 @@ use crate::documentation::{Documentation, HasDocs};
 use crate::famous_defs::FamousDefs;
 use crate::RootDatabase;
 
+#[derive(Default)]
+pub struct HoverDisplayConfig {
+    pub trait_item_display_num: Option<usize>,
+    // todo: add config for struct & enum
+}
```

Comment:
Let's put this into the hover module, and instead of modifying `Definition::label` we check if the `Definition` is a trait before calling the label function explicitly in the hover module.